src/analysis/calculus/specific_functions.lean

```diff
@@ -187,3 +194,155 @@ begin
+div_pos (exp_neg_inv_glue.pos_of_pos h) (pos_denom x)
+
+protected lemma times_cont_diff : times_cont_diff ℝ ⊤ smooth_transition :=
+smooth.div (smooth.add $ smooth.comp $ times_cont_diff_const.sub times_cont_diff_id) $
+  λ x, (pos_denom x).ne'
+
+protected lemma times_cont_diff_at {x n} : times_cont_diff_at ℝ n smooth_transition x :=
+(smooth_transition.times_cont_diff.of_le le_top).times_cont_diff_at
+
+end smooth_transition
+
+variables {E : Type*}
+
+/-- A function `f : E → ℝ` defined on a real inner product space with the following properties:
+
+- `f` is infinitely smooth on `E`;
+- `f` is positive on `ball 0 2` and equals zero otherwise;
+- `f` is equal to `1` on `closed_ball 0 1`. -/
+def smooth_bump_function [inner_product_space ℝ E] (x : E) :=
+smooth_transition (2 - ∥x∥)
+
+namespace smooth_bump_function
+
+variable [inner_product_space ℝ E]
+
+open smooth_transition
+
+lemma one_of_norm_le_one {x : E} (hx : ∥x∥ ≤ 1) : smooth_bump_function x = 1 :=
+one_of_one_le (le_sub.2 $ by { norm_num1, assumption })
+
+lemma eventually_eq_one_of_norm_lt_one {x : E} (hx : ∥x∥ < 1) :
+  smooth_bump_function =ᶠ[𝓝 x] (λ _, 1) :=
+((is_open_lt continuous_norm continuous_const).eventually_mem hx).mono $
+  λ y hy, one_of_norm_le_one (le_of_lt hy)
+
+lemma eventually_eq_one : smooth_bump_function =ᶠ[𝓝 (0 : E)] (λ _, 1) :=
+eventually_eq_one_of_norm_lt_one (by simp only [norm_zero, zero_lt_one])
+
+protected lemma times_cont_diff_at {x : E} {n} : times_cont_diff_at ℝ n smooth_bump_function x :=
+begin
+  rcases em (x = 0) with rfl|hx,
+  { exact times_cont_diff_at.congr_of_eventually_eq times_cont_diff_at_const eventually_eq_one },
+  { exact smooth_transition.times_cont_diff_at.comp x
+      (times_cont_diff_at_const.sub $ times_cont_diff_at_norm hx) }
+end
+
+protected lemma times_cont_diff {n} : times_cont_diff ℝ n (smooth_bump_function : E → ℝ) :=
+times_cont_diff_iff_times_cont_diff_at.2 $ λ x, smooth_bump_function.times_cont_diff_at
+
+protected lemma times_cont_diff_within_at {x : E} {s n} :
+  times_cont_diff_within_at ℝ n smooth_bump_function s x :=
+smooth_bump_function.times_cont_diff_at.times_cont_diff_within_at
+
+lemma nonneg (x : E) : 0 ≤ smooth_bump_function x :=
+nonneg _
+
+lemma le_one (x : E) : smooth_bump_function x ≤ 1 :=
+le_one _
+
+lemma pos_of_norm_lt_two {x : E} (hx : ∥x∥ < 2) : 0 < smooth_bump_function x :=
+pos_of_pos $ sub_pos.2 hx
+
+lemma lt_one_of_one_lt_norm {x : E} (hx : 1 < ∥x∥) : smooth_bump_function x < 1 :=
+lt_one_of_lt_one $ sub_lt.2 $ by norm_num [hx]
+
+lemma zero_of_two_le_norm {x : E} (hx : 2 ≤ ∥x∥) : smooth_bump_function x = 0 :=
+zero_of_nonpos $ sub_nonpos.2 hx
+
+lemma support_eq : function.support (smooth_bump_function : E → ℝ) = metric.ball 0 2 :=
+begin
+  ext x,
+  suffices : smooth_bump_function x ≠ 0 ↔ ∥x∥ < 2, by simpa [function.mem_support],
+  cases lt_or_le (∥x∥) 2 with hx hx,
+  { simp [hx, (pos_of_norm_lt_two hx).ne'] },
+  { simp [hx.not_lt, zero_of_two_le_norm hx] }
+end
```

Comment:
Could you put all these lemmas together with `one_of_norm_le_one`, before the smoothness assumptions?